src/main/java/com/microsoft/azure/spark/tools/http/OAuthTokenHttpObservable.java

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+package com.microsoft.azure.spark.tools.http;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.http.Header;
+import org.apache.http.client.config.CookieSpecs;
+import org.apache.http.client.config.RequestConfig;
+
+import com.microsoft.azure.spark.tools.http.AuthorizationHeader.OAuthTokenHeader;
+import com.microsoft.azure.spark.tools.utils.Lazy;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class OAuthTokenHttpObservable extends HttpObservable {
+    private final OAuthTokenFetcher accessTokenFetcher;
+    private final RequestConfig oauthDefaultRequestConfig;
+    private final Lazy<List<Header>> oAuthDefaultHeaders = new Lazy<>();
+
+    public OAuthTokenHttpObservable(final OAuthTokenFetcher toGetAccessToken) {
+        super();
+
+        this.accessTokenFetcher = toGetAccessToken;
+        this.oauthDefaultRequestConfig = RequestConfig.custom()
+                .setCookieSpec(CookieSpecs.DEFAULT)
+                .build();
+    }
+
+    public OAuthTokenHttpObservable(final String accessToken) {
```

Comment:
If we initialize the `OAuthTokenFetcher` in this way, how do we handle the token expire situation?

Reply:
It's supposed `OAuthTokenFetcher` to handle token refresh in `get()` method